xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/TemplateProviderSheet.xml

```diff
@@ -42,56 +42,52 @@
-#if ($editor == 'inline')
-  ; {{translation key="xe.templateprovider.name"/}}
-  : {{html}}&lt;input type="text" value="$!{doc.getTitle()}" name="title" /&gt;{{/html}}
-#end
-; {{translation key="xe.templateprovider.templatename"/}}
-#if ($editor == 'inline')
+{{html wiki="true"}}
+(% class="xform" %)
+(((
+  #if ($editor == 'inline')
+    ; &lt;label&gt;{{translation key="xe.templateprovider.name"/}}&lt;/label&gt;
+    : {{html}}&lt;input type="text" value="$!{doc.getTitle()}" name="title" /&gt;{{/html}}
+  #end
+  ; &lt;label&gt;{{translation key="xe.templateprovider.templatename"/}}&lt;/label&gt;
+  #if ($editor == 'inline')
+    (% class="xHint" %){{translation key="xe.templateprovider.templatename.info"/}}
+    : $doc.display('name')
+  #else
+    #set($providerName = $providerObj.getProperty('name').value)
+    : $services.localization.render($providerName)
+  #end
+  ; &lt;label&gt;{{translation key="xe.templateprovider.template"/}}&lt;/label&gt;
+  #if ($editor == 'inline')
+    : $doc.display('template')
+  #else
+    #set($templateDocName = $providerObj.getProperty('template').value)
+    : [[$templateDocName&gt;&gt;$templateDocName]]
+  #end
+  ; &lt;label&gt;{{translation key="xe.templateprovider.templatetype"/}}&lt;/label&gt;
+  #if ($editor == 'inline')
+    (% class="xHint" %){{translation key="xe.templateprovider.templatetype.info"/}}
+  #end
+  : $doc.display('type')
+  (% id="enabled-spaces" %)
+  (((
+    ; &lt;label&gt;{{translation key="xe.templateprovider.spaces"/}}&lt;/label&gt;
+    #if ($editor == 'inline')
+      (% class="xHint" %){{translation key="xe.templateprovider.spaces.info"/}}
+    #end
+    : $doc.display('spaces')
+  )))
+  ; &lt;label&gt;{{translation key="xe.templateprovider.action"/}}&lt;/label&gt;
+  #if ($editor == 'inline')
+    (% class="xHint" %){{translation key="xe.templateprovider.action.info"/}}
+  #end
+  : $doc.display('action')
 
-  {{translation key="xe.templateprovider.templatename.info"/}}
-  : $doc.display('name')
-#else
-  #set($providerName = $providerObj.getProperty('name').value)
-  : $services.localization.render($providerName)
-#end
-; {{translation key="xe.templateprovider.template"/}}
-#if ($editor == 'inline')
-  : $doc.display('template')
-#else
-  #set($templateDocName = $providerObj.getProperty('template').value)
-  : [[$templateDocName&gt;&gt;$templateDocName]]
-#end
-; {{translation key="xe.templateprovider.templatetype"/}}
-#if ($editor == 'inline')
-
-  {{translation key="xe.templateprovider.templatetype.info"/}}
-#end
-: $doc.display('type')
-(% id="enabled-spaces" %)(((
-; {{translation key="xe.templateprovider.spaces"/}}
-#if ($editor == 'inline')
-
-  {{translation key="xe.templateprovider.spaces.info"/}}
-#end
-: $doc.display('spaces')
+  #if ($xcontext.action == 'view')
+    (% class="buttonwrapper" %)[[{{translation key="xe.templateprovider.backtoadmin"/}}&gt;&gt;$xwiki.getDocument('XWiki', 'XWikiPreferences').getExternalURL('admin', 'section=Templates')]]
```

Comment:
This is a good opportunity to also fix the link problem here.

There is no reason to call `getExternalURL` here instead of `getURL`, since that will result in a an external link "marker" also being applied next to the link (next to the button in this case) and we don`t want that.

The reason this solution was chosen in the past is probably because either there was no [path:](http://platform.xwiki.org/xwiki/bin/view/Main/XWikiSyntax#HXWikiSyntax2.1LinkSpecification) option available to the link syntax (pre `xwiki/2.1`) or the writer did not know about it.

So you could also change this to `>>path:$xwiki...getURL(...)` to fix the bad display. (make sure the sheet document is set to 2.1 syntax for this to work).